### PR TITLE
Add `useAzureCommunicationChatAdapter`

### PR DESCRIPTION
**Pull request by prprabhu-ms**
Last in the sequence after #1605  and #1618 

# How Tested
Local storybook, play around with `ChatComposite` previews; Chat sample.

### CODE REVIEW

File: packages/communication-react/review/communication-react.api.md [hunk 1]
```diff
@@ -2379,12 +2379,15 @@ export type UploadFailedListener = (id: string, message: string) => void;
 // @beta
 export type UploadProgressListener = (id: string, value: number) => void;
 
-// @beta
+// @public
```
Comment (prprabhu-ms) [inline packages/communication-react/review/communication-react.api.md:2382]: Should have been @public from the start.

**Review by JamesBurnside — APPROVED** [review]

File: packages/storybook/stories/Adapters/snippets/ChatAdapterExample.snippet.tsx [hunk 2]
```diff
@@ -11,32 +11,17 @@ type ChatAdapterExampleProps = {
...
-    return () => {
-      if (chatAdapter) {
-        chatAdapter.dispose();
-      }
-    };
-  }, [props, chatAdapter]);
-
+  const credential = useMemo(() => new AzureCommunicationTokenCredential(props.accessToken), [props.accessToken]);
+  const adapter = useAzureCommunicationChatAdapter({
+    endpoint: props.endpointUrl,
```
Comment (alkwa-msft) [inline packages/storybook/stories/Adapters/snippets/ChatAdapterExample.snippet.tsx:16]: i dont think this is something we need to solve in this PR but have we consolidated on endpoint or endpointUrl? or do we need to keep them the same?

Comment (prprabhu-ms) [reply packages/storybook/stories/Adapters/snippets/ChatAdapterExample.snippet.tsx:16]: I think `endpoint` is better, but not touching that here.

File: packages/react-composites/stable-review/react-composites.api.md [hunk 3]
```diff
@@ -861,11 +861,15 @@ export type TopicChangedListener = (event: {
...
-// @beta
+// @public
 export const useAzureCommunicationCallAdapter: (args: Partial<AzureCommunicationCallAdapterArgs>, afterCreate?: ((adapter: CallAdapter) => Promise<CallAdapter>) | undefined, beforeDispose?: ((adapter: CallAdapter) => Promise<void>) | undefined) => CallAdapter | undefined;
 
 // @public
+<<<<<<< HEAD
+export const useAzureCommunicationChatAdapter: (args: Partial<AzureCommunicationChatAdapterArgs>, afterCreate?: ((adapter: ChatAdapter) => Promise<ChatAdapter>) | undefined, beforeDispose?: ((adapter: ChatAdapter) => Promise<void>) | undefined) => ChatAdapter | undefined;
+=======
 export const useAzureCommunicationCallWithChatAdapter: (args: Partial<AzureCommunicationCallWithChatAdapterArgs>, afterCreate?: ((adapter: CallWithChatAdapter) => Promise<CallWithChatAdapter>) | undefined, beforeDispose?: ((adapter: CallWithChatAdapter) => Promise<void>) | undefined) => CallWithChatAdapter | undefined;
+>>>>>>> origin/main
```
[the remark this answers is not in the document]
Comment (prprabhu-ms) [reply packages/react-composites/stable-review/react-composites.api.md:872]: Yeah, coming.

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**